src/components/formBuilder/formBuilderHelpers.ts

```diff
@@ -201,19 +202,14 @@ export const validateNextPropertyName = (
     return "Name must not contain periods.";
   }
 
-  if (
-    schema.properties &&
-    Object.prototype.hasOwnProperty.call(schema.properties, nextPropertyName)
-  ) {
-    return `Name must be unique. Another property "${
-      // eslint-disable-next-line security/detect-object-injection -- checked with hasOwnProperty
-      (schema.properties[nextPropertyName] as Schema).title
-    }" already has the name "${nextPropertyName}".`;
+  if (hasOwnProp(schema.properties, nextPropertyName)) {
+    const { title } = getOwnProp(schema.properties, nextPropertyName) as Schema;
+    return `Name must be unique. Another property "${title}" already has the name "${nextPropertyName}".`;
   }
 
   if (
-    // Checked Own Properties already.
     // If the property with nextPropertyName is defined nevertheless, there's something wrong with the new name.
+    // eslint-disable-next-line security/detect-object-injection -- It doesn't work with hasOwnProp/getOwnPropaa
     typeof schema.properties?.[nextPropertyName] !== "undefined" ||
```

Comment:
Neither one worked and I'm not sure why (it's setting prototype properties?)

```suggestion
    typeof getOwnProp(schema.properties, nextPropertyName) !== "undefined" ||
```

```suggestion
    !hasOwnProp(schema.properties, nextPropertyName) ||
```

This may suggest that the rest of the changes might also not be equivalent or "safe". My smoke tests passed without errors though.

I'm open to splitting this PR into 2/4 pieces and merging them over time so should we have issues we don't have to revert all of them.